File: exercise_three/scripts/pub_test.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
import rospy
from exercise_three.msg import addend

def talker():
    # 初始化节点
    rospy.init_node('topic_pub', anonymous=False)
    # 打印输出日志消息
    rospy.loginfo('topic_pub node is Ready!')
    # 创建发布RobotInfo消息到话题/robot_info的句柄(发布端)pub
    pub = rospy.Publisher('/addend', addend, queue_size=10)
    # 创建了addend消息的对象msg,并赋值
    msg = addend()
    msg.x = 1
    msg.y = 3
    msg.z = 2
    # 创建rate对象,设置频率为5Hz,用于循环发布
    rate = rospy.Rate(5)
    # 节点关闭前一直循环发布消息
    while not rospy.is_shutdown():
        # 发布消息
        pub.publish(msg)
        # 按照循环频率延时
        rate.sleep()

if __name__ == '__main__':
    talker()

File: exercise_three/src/pub_test.cpp
#include "ros/ros.h"
#include "exercise_three/addend.h"	

int main(int argc,char **argv){
  //初始化ROS节点
  ros::init(argc, argv, "pub_test");
  // 创建一个节点句柄（NodeHandle）对象nh
  ros::NodeHandle nh;
  // 打印输出日志消息
  ROS_INFO("pub_test node is Ready!");
  // 创建Publisher对象pub,用于向话题/addend发布addend消息
  ros::Publisher pub = nh.advertise<exercise_three::addend>("/addend", 10);
  // 创建了addend的对象msg，并对msg里面的部分成员进行赋值
  exercise_three::addend msg;
  msg.x = 1;
  msg.y = 2;
  msg.z = 3;
  // 创建rate对象,设置频率为10Hz,用于循环发布
  ros::Rate rate(10);
  //循环发布
  while(ros::ok()){
    //打印输出xyz的数值
    ROS_INFO("x = %f, y = %f, z=%f", msg.x , msg.y ,msg.z);
    pub.publish(msg);
    // 按照循环频率延时
    rate.sleep();
  }
  return 0;
}
